include/rocksdb/stats_history.h

```diff
@@ -11,14 +11,28 @@
 #include <map>
 #include <string>
 
-// #include "db/db_impl.h"
 #include "rocksdb/statistics.h"
 #include "rocksdb/status.h"
 
 namespace rocksdb {
 
 class DBImpl;
 
+// StatsHistoryIterator is the main interface for users to programmatically
+// access statistics snapshots that was automatically stored by RocksDB.
+// Depending on options, the stats can be in memory or on disk.
+// The stats snapshots are indexed by time that they were recorded, and each
+// stats snapshot contains individual stat name and value at the time of
+// recording.
+// Example:
+//   std::unique_ptr<StatsHistoryIterator> stats_iter;
+//   db->GetStatsHistory(0 /* start_time */, env->NowMicros() /* end_time*/,
+//                       &stats_iter);
+//   for (; stats_iter->Valid(); stats_iter->Next()) {
+//     uint64_t stats_time = stats_iter->GetStatsTime();
+//     auto stats_map = stats_iter->GetStatsMap();
```

Comment:
As a good example, follow the rule to spell out the type?
Also, here no "&" is given, so I believe it is a copy. Do we want to give an inefficient example?